policies/user_policy.rb

```diff
@@ -0,0 +1,31 @@
+class UserPolicy
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z/i
+
+  def initialize(params)
+    @params = params
```

Comment:
There's no point of assigning `params` to an instance variable and then using memoization to retrieve password & email since accessing hash value by key is an `O(1)` operation. Do something like this
```ruby
def initialize(params)
  @email    = params[:email]
  @password = params[:password]
end
```